pkgs/servers/wesher/default.nix

```diff
@@ -0,0 +1,29 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "wesher";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "costela";
+    repo = "wesher";
+    rev = "v${version}";
+    sha256 = "sha256-EIajvcBhS5G9dJzRgXhnD1QKOAhmzngdyCU4L7itT8U=";
+  };
+
+  vendorSha256 = "sha256-BZzhBC4C0OoAxUEDROkggCQF35C9Z4+0/Jk0ZD8Hz1s=";
+
+  ldflags = [
+    "-s" "-w" "-X main.version=${version}"
+  ];
+
+  meta = with lib; {
+    description = "Wireguard overlay mesh network manager";
+    homepage = "https://github.com/costela/wesher";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ tylerjl ];
```

Comment:
```suggestion
    maintainers = with maintainers; [ tylerjl ];
    platforms   = platforms.linux;
```

Reply:
@SuperSandro2000 so I tried building against `GOOS=darwin` on varying go versions (1.17, 1.18, 1.19) on different revisions of the upstream code but it looks like there isn't a combination that compiles successfully for darwin - coupled with the fact that the [releases page](https://github.com/costela/wesher/releases/tag/v0.2.6) only lists Linux artifacts, I think scoping `wesher` down solely to `platforms.linux` is probably the right choice. Would this change be acceptible?

Reply:
@tylerjl please do mark it for `platforms.linux`

Reply:
Yeah, we can do that but I am pretty sure that bumping x/sys fixes the current error. Did you try that and there where others?

Reply:
@balsoft thanks, it seems like the right approach to me :+1: 

@SuperSandro2000 I bumped the dependencies but ended up chasing more errors - I may head to the upstream repo and ask for a new release to get newer deps/updated changes which may help here.